### ISSUE TITLE
Modsuit springlock moduals do not kill when sprayed or hit or swim in water

### ISSUE THREAD

**Issue by tobjv** [Skyrat-SS13/Skyrat-TG#13177]
Reporting client version: 514.1582

## Round ID:
5186

## Testmerges:
- [Goofsec 2.0 aka "I guess we're going back to bullets in Security": Security Pistol w/ 6 Ammo Types, Warden Dispatch Console](https://github.com/Skyrat-SS13/Skyrat-tg/pull/12949)
- [[MODULAR] Status Indicators with Preference](https://github.com/Skyrat-SS13/Skyrat-tg/pull/12529)
- [Database revision revert](https://github.com/Skyrat-SS13/Skyrat-tg/pull/10823)

**Comment by GoldenAlpharex** [comment]
@tobjv So basically they were still working even when you were in contact with water (or any other reagent), just to be clear?

**Comment by tobjv** [comment]
> @tobjv So basically they were still working even when you were in contact with water (or any other reagent), just to be clear?

yep, they worked at speeding up the power up and down phase of deployment with no downside

**Comment by Iamgoofball** [comment]
@tobjv can you reproduce this on /tg/

**Comment by Iamgoofball** [comment]
you probably spawned the contractor variant that @Zonespace27 added that does not kill you
the ones that kill you are:
/obj/item/mod/module/springlock
/obj/item/mod/module/springlock/bite_of_87
the one that doesn't kill you:
/obj/item/mod/module/springlock/contractor
https://github.com/Skyrat-SS13/Skyrat-tg/blob/master/modular_skyrat/modules/contractor/code/items/modsuit/modules.dm#L213

**Comment by Iamgoofball** [comment]
additionally, EMPs don't impact any version of the springlock module, not sure why that's in the issue report either
edit: they got pranked by the bite_of_87 subtype making itself disguised as a dna lock

**Comment by tobjv** [comment]
> additionally, EMPs don't impact any version of the springlock module, not sure why that's in the issue report either

because the text on the dna one says it is effected by EMP. that's why I tested water AND EMP's.

I must have not been clear, I tested it on all three mods spawnable via the admin panel.

the examine of it is `That's MOD DNA lock module.
--------------------------------------------------------------------------------
A module which engages with the various locks and seals tied to the suit's systems, enabling it to only be worn by someone corresponding with the user's exact DNA profile; however, this incredibly sensitive module is shorted out by EMPs. Luckily, cloning has been outlawed.
It is a normal-sized item.`

It states the DNA lock is shorted by them. Why would that not be "triggers death?"

I hit all three of them with water and EMPS to no effect.

No I did not test it on TG, I don't have admin powers there. Yes I could have started a single player to find out but choose to spend my time admin the rounds instead. I post my bugs here as this is a issue I found on Skyrat. Someone who wants to can go test it on TG.
